### ISSUE TITLE
[Infra] 모델 비교 판정이 임계값 통과 여부만으로 결정되고, 신뢰도·다중 신호 불일치가 구조화되지 않음

### ISSUE THREAD

**Issue by bbungjun** [SKYAHO/Autoresearch#425]
## 배경

Capability probe round_002(MLP vs LightGBM)에서 실제로 벌어진 상황입니다.

> **책임 분리**: 다중 시드 반복 학습(시드 스윕 실행 경로) 자체는 #407이 담당합니다. 이 이슈는 그 결과를 포함한 **여러 판정 신호를 어떻게 구조화해서 남길지**만 다룹니다. 경위는 코멘트 참고.

## 문제 — 임계값 통과 여부만으로 판정이 확정되고, 다중 신호 불일치가 구조화되지 않음

공식 프로토콜(고정 시드) 기준 MLP가 LightGBM보다 test ROC-AUC +2.63pp 높아 "채택" 판정을 받았습니다. 그런데 데이터 split은 고정한 채 MLP 가중치 초기화 시드만 5개로 바꿔보니 2개만 baseline을 넘겼고 5개 평균(0.8933)은 baseline(0.9024)보다 낮았습니다. 보조지표(PR-AUC/LogLoss/Brier)는 모든 시드에서 일관되게 LightGBM 우위였습니다. (이 시드 스윕 자체를 파이프라인이 지원하게 만드는 것은 #407의 범위입니다.)

같은 라운드 안에서 (a) 오프라인 주지표, (b) 시드 민감도 진단, (c) 정책시뮬레이션(실 LLM) 이렇게 세 신호가 나올 수 있는데, round_002에서 이 셋이 서로 다른 방향(정책시뮬레이션은 주지표와 같은 방향, 시드민감도와는 반대 방향)을 가리켰습니다. 지금은 이걸 자유 서술로만 기록하지, `verdict` 하나로 뭉뚱그리면 안 된다는 걸 강제하는 스키마가 없습니다.

## 왜 중요한가

자율 에이전트가 승격 후보를 판단할 때 "임계값 통과"만 보고 자동으로 강한 확신을 가지면 위험합니다. 실제로 이번 실험은 격차가 임계값의 2.6배였음에도 결과가 신뢰할 수 없는 수준이었습니다.

## 제안 범위

- 판정 산출물 스키마에 `verdict`와 별개로 `confidence`/`robustness_note` 필드, 신호별 `direction_vs_*` 필드 표준화
- "기각"·"낮은 신뢰도"를 정상 종료 경로로 다루는 설계 가이드

## 완료 조건

- [ ] 판정 산출물 스키마에 `confidence`/`robustness_note` 필드가 추가된다
- [ ] 신호별(`direction_vs_offline_metric`/`direction_vs_seed_sensitivity`/`direction_vs_policy_sim` 등) 방향 일치 여부가 구조화되어 남는다
- [ ] "기각"·"낮은 신뢰도" 결과가 실패가 아닌 정상 종료 경로로 문서화된다

## 관련

- #407 — 다중 시드 반복 학습 자동화(이 이슈가 소비하는 입력을 만드는 이슈, 책임 분리 경위는 코멘트 참고)

**Comment by waieiches** [comment]
## 겹치는 부분 대조 (전체는 #403에)

> **정정**: 처음엔 문제1이 #407과 "정면으로 같은 요구"라고 적었는데, 다시 보니 **중복이 아니라 순서 관계**입니다.

- **#407** : 시드 목록으로 반복 학습하는 **기능**을 만드는 것
- **이 이슈 문제1** : 그 기능을 **언제 자동으로 발동할지** 정하는 것

#407이 없으면 이 이슈 문제1은 만들 수가 없습니다. 그래서 합치는 게 아니라 **#407 먼저, 이 이슈 나중**이 맞습니다. 앞서 "#407로 옮기고 이 이슈는 문제2만 남기자"고 적은 건 취소하겠습니다.

### 다만 문제1에 붙은 실측은 #407의 근거로 값이 큽니다

시드 5개 중 2개만 baseline을 넘었고, 평균은 baseline보다 낮았고, 다른 지표들은 모든 시드에서 반대 방향이었다는 부분입니다.

#407은 그동안 #396의 "+0.0019로 기각, 시드 하나로만 확인" 사례가 근거였는데, 이제 **실제로 판정이 뒤집힌 사례**가 생겼습니다. 이 내용을 #407에도 인용해 두었습니다.

### 문제2는 겹치는 게 없습니다

여러 신호가 서로 다른 방향을 가리킬 때 어떻게 기록할지(`verdict` 말고 `confidence`/`robustness_note`, 신호별 `direction_vs_*`)는 #407에 없는 새 요구입니다. 따로 진행하셔도 됩니다.

**Comment by bbungjun** [comment]
## 책임 분리 정리

#407과 범위가 겹쳐 있어 정리합니다.

- **다중 시드 반복 학습(시드 스윕 실행 경로) 자체는 #407이 담당**합니다. 이 이슈의 "문제 1" 제안(bootstrap CI/다중 시드 재실행 자동 요구)은 #407의 제안 범위와 동일한 요구라 중복 발행이었습니다.
- 이 이슈는 **판정 신호를 어떻게 구조화해서 남길지**(주지표·시드 민감도·정책시뮬레이션 세 신호의 방향 불일치를 `verdict` 하나로 뭉개지 않고 `confidence`/`direction_vs_*` 필드로 표준화)만 남깁니다.
- 즉 #407이 만드는 "시드별 반복 결과"를 **입력**으로 받아, 이 이슈가 "판정 스키마"라는 **출력**을 만드는 관계입니다. 병렬 작업 시 서로 다른 산출물이라 겹치지 않습니다.

본문을 좁혀서 다시 작성합니다.

**Comment by waieiches** [comment]
## 실측 근거 — 3 seed에서는 신뢰구간이 0을 포함합니다 (2026-08-09)

이 이슈가 지적한 *"임계값 통과 여부만으로 결정되고 신뢰도가 구조화되지 않는다"*는 문제에, 실제 파이프라인 완주 데이터가 나와 근거로 남깁니다.

### 데이터

같은 가설을 두 번 돌린 실험 2건입니다(issue #634 · #635). 가설·코드 변경·지표가 **소수점 16자리까지 동일**하게 재현됐습니다.

```
가설        LightGBM learning_rate 0.05 → 0.03
변경        src/models/lgbm_model.py 2줄 + src/pipeline/config.yaml 1줄
base_dev_sha  8242d3b6 (양쪽 동일)
seeds         [42, 43, 44]

baseline  roc_auc = 0.6272648084197142
candidate roc_auc = 0.6358108352753750
paired    delta mean = 0.008546026855660749
          standard_error = 0.0021301057134867915
```

### 문제

3 seed면 자유도 2, `t_critical ≈ 4.30`입니다.

```
95% CI ≈ 0.008546 ± 4.30 × 0.002130
       ≈ [-0.0007, +0.0177]      ← 0을 포함합니다
```

**delta 평균은 +0.0085로 개선처럼 보이지만 통계적으로는 유의하지 않습니다.** 표준오차 자체는 작은데(0.0021) `t_critical`이 4.30이라 구간이 벌어집니다.

seed를 10개로 늘리면 자유도 9·`t_critical ≈ 2.26`이 되어 같은 표준오차에서도 구간이 절반 이하로 좁아집니다. 계획 문서(`docs/plans/2026-08-07-experiment-execution-enablement.md` Stage 1-4)가 후속 항목으로 남겨둔 내용인데, **이제 그 판단을 뒷받침할 실측이 생겼습니다.**

### 왜 이 이슈에 남기는가

현재 `PASSED`는 통계적 판정이 아니라 **"실험이 완주하고 결과가 나왔다"**로 재정의됐습니다. 그래서 위 실험도 `PASSED`입니다 — **신뢰구간이 0을 포함한다는 사실이 상태 어디에도 드러나지 않습니다.**

이 이슈가 말한 "신뢰도가 구조화되지 않은" 상태가 지금 그대로입니다. 지표는 `metric_summary`에 실려 있으므로 **판정을 되살릴 때 구간 계산에 필요한 재료(평균·표준오차·seed 수)는 이미 확보돼 있습니다.**

### 시간 예산 참고

seed 상향의 비용은 계획 문서에 실측이 있습니다 — 3일치 스냅샷·1코어 기준 학습 1회 **19.5초**, seed 10개면 두 조건 합쳐 390초입니다. 이번 실험의 Job 총 소요는 31~33분이었고 `activeDeadlineSeconds` 상한은 3600초이므로 **여유가 있습니다.**

다만 상한 인상 논의(`SKYAHO/Autoresearch-infra#603`)와 함께 보면 총 예산이 달라지므로 그쪽과 같이 판단하는 편이 낫습니다.

### 관련

- `#472` — 재학습·승격 게이트 정책 (같은 축의 결정)
- `SKYAHO/Autoresearch-infra#603` — activeDeadlineSeconds 상한 재검토
- 계획 `docs/plans/2026-08-07-experiment-execution-enablement.md` Stage 1-4 후속 항목
